mfr/extensions/zip/render.py

```diff
@@ -0,0 +1,30 @@
+import os
+import zipfile
+
+from mako.lookup import TemplateLookup
+
+from mfr.core import extension
+
+
+class ZipRenderer(extension.BaseRenderer):
+
+    TEMPLATE = TemplateLookup(
+        directories=[
+            os.path.join(os.path.dirname(__file__), 'templates')
+        ]).get_template('viewer.mako')
+
+    def render(self):
+        zip_file = zipfile.ZipFile(self.file_path, 'r')
+
+        return self.TEMPLATE.render(zipped_filenames=self.format_zip(zip_file))
+
+    def format_zip(self, zip_file):
+        return '<br>'.join(zip_file.namelist())
```

Comment:
I'm having a hard time coming up with an actionable example, but since users decide the file name, please markup-escape these strings before rendering.  Bad example: if a user calls their file `<b>foo</b>`  the tags should not be interpreted as markup. Sample usage: https://github.com/CenterForOpenScience/modular-file-renderer/blob/81131736cdd44d27a5fece6d62bac1f52f017abe/mfr/core/provider.py#L23